webgme.js

```diff
@@ -46,8 +47,9 @@ function addToRequireJsPaths(gmeConfig) {
                 try {
                     items = fs.readdirSync(basepaths[j]);
                     if (items.indexOf(componentNames[i]) !== -1) {
-                        componentPaths[componentName + '/' + componentNames[i]] = path.relative(requireJsBase,
-                            path.resolve(basepaths[j]));
+                        componentPath = path.relative(requireJsBase, path.resolve(basepaths[j]))
+                            .replace(/^client\/js/, 'js');
+                        componentPaths[componentName + '/' + componentNames[i]] = componentPath;

```

Comment:
This was added because currently all components using this function (the components with `basePaths` in the config) have the same path on the filesystem as their path hosted on the server (eg, `plugin`, `common` as both are in `src/`).

However, when we are adding base paths for `layout`, the path on the server doesn't match the path on the file system. That is, if a component has `basePaths` set in the config but exists in `src/client/js` on the filesystem, it's path on the server is just `js/rest/of/path`.

Anyway, this should not cause problems with any other components unless we decide to add a `client/js` on the server which does not correspond to the `client/js` on the file system.